### PR TITLE
add convenience methods and expose Reflex.UI

### PR DESCRIPTION
**Pull request by tippenein**
There is an implied pattern of <element>' meaning 'the simple version' (`h1' "Header Title"`) where it only takes the text input or nothing instead of `Attrs -> m a -> m a`. This seems "ok" at best. I'd be open to other suggestions. I also changed the infix tuple-maker `=:` to `==:` in order to avoid collisions with reflex-dom. I'm also not wedded to this because this will most likely be imported qualified, so it may not be an issue.

### CODE REVIEW

**Comment by TikhonJelvis** [comment]
Some good ideas here. It brings up some design questions worth talking about.

One is how complete we can expect `reflex-ui` to be. My thought was to develop a framework you could use without accessing `reflex-dom` functions directly—does this sound like a usable goal? If it is, we don't have to worry about shadowing things in `reflex-dom`. We can always re-export selected operations if we think people will need them.

The `'` versions of elements like `h1` are another interesting design choice. Could this be done with `OverloadedStrings`? If it's possible, that would be a good compromise: you could include literal text directly and you would only have to wrap a text into an element if it was named—which seems fine.

The problem I have with these special convenience functions is that they quickly proliferate. On the one hand, I agree it's nice to not wrap things in `text` explicitly; on the other, we might also want similar convenience functions for dynamic text (ie wrapping with `dynText` instead of `text`). The `reflex-dom` API does this, and I think that's a big part of why it's confusing and hard to follow: it's just too redundant.

Personally, I would want to hold off on the `'` functions until we have a compelling story for them overall. One thing I could see doing is creating an `HTML` class similar to `Show`; this would let us have a single function that accepted any type that could be canonically rendered as an element, including `Text` and `String`. What do you think?

**Comment by tippenein** [comment]
I agree with pretty much all of this. The `'` methods were not good to begin with, but for lack of a better default I included them. I'm mostly not sure what an `HTML` class interface would look like or how we could remove boilerplate with it.

**Comment by TikhonJelvis** [comment]
I'm imagining the following:

```
class ToHTML a where
  toHTML :: MonadWidget t m => a -> m ()
```

The details and naming is rough, but the idea is to have something like `Show` but for DOM elements.

Then we could do stuff like:

```
instance ToHTML Text where
  toHTML = text

h1 :: (ToHTML a, MonadWidget t m) => a -> m ()
```

I don't know how well this would play out in practice, but it's probably worth a try. We could also have a default instance for things already in a `MonadWidget t m`, but that might work poorly:

```
instance MonadWidget t m => ToHTML (m a) where
  toHTML = void . id
```

How does that sound?
